src/storage.h

```diff
@@ -0,0 +1,253 @@
+
+  /// @brief move ready resources from processing to ready at a certain time
+  /// @param time the time of interest
+  void ReadyMatl_(int time);
+
+  /* --- Module Members --- */
+
+  #pragma cyclus var {"tooltip":"input commodity",\
+                      "doc":"commodities accepted by this facility",\
+                      "uilabel":"Input Commodities",\
+                      "uitype":["oneormore","incommodity"]}
+  std::vector<std::string> in_commods;
+
+  #pragma cyclus var {"default": [],\
+                      "doc":"preferences for each of the given commodities, in the same order."\
+                      "Defauts to 1 if unspecified",\
+                      "uilabel":"In Commody Preferences"}
+  std::vector<double> in_commod_prefs;
+
+  #pragma cyclus var {"tooltip":"output commodity",\
+                      "doc":"commodity produced by this facility. Multiple commodity tracking is"\
+                      " currently not supported, one output commodity catches all input commodities.",\
+                      "uilabel":"Output Commodities",\
+                      "uitype":["oneormore","outcommodity"]}
+  std::vector<std::string> out_commods;
+
+  #pragma cyclus var {"default":"",\
+                      "tooltip":"input recipe",\
+                      "doc":"recipe accepted by this facility, if unspecified a dummy recipe is used",\
+                      "uilabel":"Input Recipe",\
+                      "uitype":"recipe"}
+  std::string in_recipe;
+
+  #pragma cyclus var {"default": 0,\
+                      "tooltip":"residence time (timesteps)",\
+                      "doc":"the minimum holding time for a received commodity (timesteps).",\
+                      "units":"time steps",\
+                      "uilabel":"Residence Time"}
+  int residence_time;
+
+  #pragma cyclus var {"default": 1e299,\
+                     "tooltip":"throughput per timestep (kg)",\
+                     "doc":"the max amount that can be moved through the facility per timestep (kg)",\
+                     "uilabel":"Throughput",\
+                     "units":"kg"}
+  double throughput;
+
+  #pragma cyclus var {"default": 1e299,\
+                      "tooltip":"maximum inventory size (kg)",\
+                      "doc":"the maximum amount of material that can be in all storage buffer stages",\
+                      "uilabel":"Maximum Inventory Size",\
+                      "units":"kg"}
+  double max_inv_size; 
+
+  #pragma cyclus var {"default": True,\
+                      "tooltip":"Bool to determine how Storage handles batches",\
+                      "doc":"Determines if Storage will divide resource objects. "\
+                      "If true, batches are handled as discrete quanta, neither split nor combined. "\
+                      "Otherwise, batches may be divided during processing. Default to true (discrete)",\
+                      "uilabel":"Batch Handling"}
+  bool batch_handling;                    
```

Comment:
It looks like some incorrect assumptions are made about the behavior of the buy/sell policies w.r.t. discrete/continuous material handling.  Roughly speaking, the policies as configured will always operate in continuous fashion - allowing resource exchange to resolve trades that require splitting/combining materials.  We should put a TODO comment in the code about fixing this - perhaps in the cc file.  And we should document here that continuous material handling is used when trading, but internal handling honors this setting.  And could we rename this to "discrete_handling"?